tests/RobotFramework/tests/configuration/lock_file.robot

```diff
@@ -61,3 +61,13 @@ Switch off lock file creation
     #Check that no lock file is created
     File Should Not Exist    /run/lock/tedge-agent.lock
     File Should Not Exist    /run/lock/tedge-mapper-c8y.lock
+
+*** Keywords ***
+
+Stop/Start Service
+    [Arguments]    ${name}
+    # Note: Use stop/start service rather than restart as 'systemctl restart <service>'
+    # does not wait for all file descriptors to be flushed before returning, https://man7.org/linux/man-pages/man1/systemctl.1.html
+    # See https://github.com/thin-edge/thin-edge.io/issues/2087 for more details
+    Stop Service    ${name}
```

Comment:
How about waiting for a few seconds before starting, to make sure that the OS flushes all the FDs properly.

Reply:
Using stop/start is a cleaner solution as systemd will wait accordingly...with a sleep it is hard to know how much to sleep for as it the performance of the ci runner is very unpredicatable.